lib/phoenix/pubsub/pg2.ex

```diff
@@ -83,6 +98,16 @@ defmodule Phoenix.PubSub.PG2 do
         Supervisor.child_spec({Phoenix.PubSub.PG2Worker, {name, group}}, id: group)
       end
 
+    children =
+      case System.get_env("PHX_CLUSTER_DNS") do
```

Comment:
We really shouldn't be relying on env vars down here. Also, the cluster name is not related to PubSub, so I would say this functionality is out.

My suggestion is to add `Phoenix.PubSub.DNSCluster` as a service and have your Phoenix endpoint start it based on a config:

    config :my_app, MyApp.Endpoint, dns_cluster: true

Reply:
It's not "related", but also the pg adapter is not useful if you aren't clustered, so it's still pretty well related. I'm fine with the endpoint starting it as well, but we have a problem in that the pubsub adapter is started above the endpoint for good reasons, so we'd be starting pubsub before clustering, which seems backwards, so I'm not sure?

Reply:
So to put another way, I'm convinced we need a knob free way to get out of the box clustering, but if we don't put in the pubsub tree we don't have any other option to start it in the correct order, it short of adding it explicitly adding it to the children in `phx.new`